### PR TITLE
Handling images

### PR DESCRIPTION
**Pull request by mhz-edu**
Added model, controller, routes to habdle images.

### CODE REVIEW

File: src/controllers/images.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { RequestHandler } from 'express';
+const { ExifImage } = require('exif');
```
Comment (irenlian) [inline src/controllers/images.ts]: Imports should be consistent.

File: src/controllers/images.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import Image from '../models/image';
+import coordinatesConverter from '../util/coordinatesConverter';
+
+export const postImage: RequestHandler = (req, res) => {
+  let albumId = parseInt(req.params.albumId);
+  new Promise(function (resolve, reject) {
+    new ExifImage(
+      { image: req.file!.buffer },
+      (error: Error, exifData: any) => {
```
Comment (irenlian) [inline src/controllers/images.ts]: For each external library in TS, you need an additional types library.
So this `any` will be changed
https://www.npmjs.com/package/@types/exif

File: src/util/coordinatesConverter.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  if (gpsParams.GPSLatitudeRef === 'S') {
+    latSign = -1;
+  }
+  if (gpsParams.GPSLongitudeRef === 'W') {
+    lonSign = -1;
+  }
+  let [latD, latM, latS] = gpsParams.GPSLatitude;
+  let [lonD, lonM, lonS] = gpsParams.GPSLongitude;
+  const latDegree = latSign * (latD + latM / 60 + latS / 3600);
+  const lonDegree = lonSign * (lonD + lonM / 60 + lonS / 3600);
```
Comment (irenlian) [inline src/util/coordinatesConverter.ts]: I'd move `3600` and `60` to constants, so they would have names and be more readable.

File: src/util/coordinatesConverter.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const coordinatesConverter = (gpsParams: any) => {
```
Comment (irenlian) [inline src/util/coordinatesConverter.ts]: The input type will be specified with the type library from the comment above.

File: src/routes/images.ts [hunk 5]
```diff
@@ -1,31 +1,19 @@
...
-    (result) => res.json(result),
-    (error) => res.json(error)
-  );
-});
+imagesRouter.get('/api/albums/:albumId/images', getAlbumImages);
+
+imagesRouter.post(
+  '/api/albums/:albumId/images',
+  upload.single('upfile'),
+  postImage
```
Comment (irenlian) [inline src/routes/images.ts:27]: Do you have any restrictions on the files? (type, extension, size)

**Review by irenlian — COMMENTED** [review]
I'm going to leave a few considerations regarding the project overall. You don't need to fix them now, but in the separate PRs. Most of them are just about best practises or developer experience, so no worries.
- take a look into nodemon
- please, fill Readme file with the sections: description, backlog, how to run, existing feature list
- add `.env` to `.gitignore`
- add validation to POST requests
- add graceful shutdown
- some parts of routes may be combined - like `api/` that could be separated as a common part
- switching between DB versions should be using `NODE_ENV` env variable

And some notes regarding images are below:

File: src/controllers/images.ts (hunk 1 above)
Comment (mhz-edu) [reply src/controllers/images.ts]: updated with types

File: src/controllers/images.ts (hunk 2 above)
Comment (mhz-edu) [reply src/controllers/images.ts]: updated with types

File: src/util/coordinatesConverter.ts (hunk 4 above)
Comment (mhz-edu) [reply src/util/coordinatesConverter.ts]: updated with types

File: src/util/coordinatesConverter.ts (hunk 3 above)
Comment (mhz-edu) [reply src/util/coordinatesConverter.ts]: moved to constants

File: src/routes/images.ts (hunk 5 above)
Comment (mhz-edu) [reply src/routes/images.ts:27]: added restrictions on type and size

**Review by irenlian — APPROVED** [review]

**Pull request merged**